Azure Services/Azure Monitor/Agents/Migration Tools/DCR Config Generator/WorkspaceConfigToDCRMigrationTool.ps1

```diff
@@ -655,6 +823,43 @@ function Get-DataFlows
     return $dataFlows
 }
 
+function Get-CustomLogStreamDeclarations
+{
+    param (
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+
+    $dataSourceType = "CustomLog"
+    $workspaceDataSourceList = Get-AzOperationalInsightsDataSource -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName -Kind $dataSourceType
+    $streamDeclarations = @{}
+
+    foreach($dataSource in $workspaceDataSourceList)
+    {
+        if($dataSource.Properties -ne $null -and $dataSource.Properties.customLogName -ne $null)
+        {
+            $streamName = "Custom-" + $dataSource.Properties.customLogName
+            $streamDeclarations[$streamName] = @{
+                "columns" = @(
+                    @{
+                        "name" = "TimeGenerated";
+                        "type" = "datetime";
+                    },
+                    @{
+                        "name" = "RawData";
+                        "type" = "string";
+                    }
+                )
+            }
+        }
+        else {
+            Write-Host "Error: Unable to get stream name for Custom Log data source."
+        }
+    }
+    # TODO: Ask if it's okay to have streamDeclaration as empty
+    return $streamDeclarations
```

Comment:
In case of an error, is it okay if `$streamDeclarations` is returned as empty? Or should we think of another default?

Reply:
As long as the payload is passing DCR PUT Validations, this should be fine. Let's that scenario (by manually setting streamDeclarations to empty)

Reply:
Yes I verified that the DCR deployment goes through